### PR TITLE
mgr/cephadm: osd should not be zap when it is running

### PR DESCRIPTION
**Pull request by strenuous-life**
When osd is running, we exec the command of "ceph orch device zap".
Then ceph-volume will exec dd command which cause the osd of lvs to be destroyed.
If the osd of systemctl unit restarts, it will be unable to boot.

Signed-off-by: jianglong01 <jianglong01@qianxin.com>

Fixes: https://tracker.ceph.com/issues/51028

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -1976,6 +1976,23 @@ def get_inventory(self, host_filter: Optional[orchestrator.InventoryFilter] = No
...
+            raise OrchestratorError('Zap failed: ceph-volume lvm list %s' % '\n'.join(path))
+        osd_info = json.loads('\n'.join(out))
+
+        for osd_id in osd_info.keys():
+            out, err, code = CephadmServe(self)._run_cephadm(
+                host, 'osd.', 'unit',
+                ['is-active', '--name', 'osd.%s' % osd_id],
+                error_ok=True)
+            if code == 0:
+                raise OrchestratorError('The systemctl unit of osd is active. Rm osd first')
```
[the remark this answers is not in the document]
Comment (strenuous-life) [reply src/pybind/mgr/cephadm/module.py:1994]: zap_device means zap path which is specified disk, for example, /dev/sdb.
However, zap-osds of cehpadm means clear up all disk and it is not suitable for this scene.

[the remark this answers is not in the document]
Comment (strenuous-life) [reply src/pybind/mgr/cephadm/module.py:1994]: zap_device means zap path which is specified disk, for example, /dev/sdb.
However, zap-osds of cehpadm means clear up all disk and it is not suitable for this scene.

[the remark this answers is not in the document]
Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/module.py:1994]: Ok, seems that we need to introduce a new comand `cephadm zap-disk <device>` than.

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-09-23_14:38:36-rados:cephadm-wip-swagner-testing-2021-09-23-1100-distro-basic-smithi/

**Review by sebastian-philipp — APPROVED** [review]

**Pull request merged**